src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -20,36 +21,49 @@ public class DeleteCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Parameters: INDEX [OPTIONAL INDEXES] (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deletion successful!";
+    public static final String NO_PERSON_TO_DELETE_ERROR = "There is no student to delete.";
 
-    private final Index targetIndex;
+    private final List<Index> targetIndexes;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    /**
+     * Creates a DeleteCommand to delete the specified {@code Person}(s).
+     *
+     * @param targetIndexes of the person(s) to delete.
+     */
+    public DeleteCommand(List<Index> targetIndexes) {
+        Collections.sort(targetIndexes);
+        this.targetIndexes = targetIndexes;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (model.getFilteredPersonList().size() == 0) {
+            throw new CommandException(NO_PERSON_TO_DELETE_ERROR);
         }
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
-        model.deletePerson(personToDelete);
-        ClassStorage.removeExistingClass(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+        for (int i = targetIndexes.size() - 1; i >= 0; i--) {
+            Index targetIndex = targetIndexes.get(i);
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
What would we do if one of the indexes is out of the range, ie invalid? By this implementation, we are deleting all the persons with indexes that are valid and before the invalid index but the error message would pertain. I think it would make more sense if we check whether all indexes are valid first and raise the exception (reject the input) beforehand, rather than execute until the invalid index?

Reply:
This can be potentially relevant to the overzealous validation so I think it would be better for a group decision here. What do yall think? @gnahzeus @seanflyyy @cadencjk @sjoann

Reply:
Good point raised! Actually, the current implementation will check for all indexes, and if any of the indexes are invalid, it will return the error message without deleting the valid indexes.

Reply:
Thanks for the prompt! I am not sure what you mean exactly so I will try to clarify with a few testcases.
The current implemention:
1. The system goes through all the inputs first, as long as there is a non positive number (<= 0), it will return this error message:
<img width="465" alt="Screenshot 2022-10-23 at 3 44 54 PM" src="https://user-images.githubusercontent.com/81757201/197380540-88e8eab7-e8ac-47ac-8783-dacea5d29bc8.png">

<img width="470" alt="Screenshot 2022-10-23 at 3 47 26 PM" src="https://user-images.githubusercontent.com/81757201/197380645-71490d78-6ae4-4a62-91a6-dc20fb882527.png">

2. Once the parser knows that there is no non positive number in the input, it checks if there are any non positive numbers exceeding the list (> size of list). A different error message will show. Even if a correct index is included, it will not perform any deletion.

<img width="486" alt="Screenshot 2022-10-23 at 3 49 52 PM" src="https://user-images.githubusercontent.com/81757201/197380737-7a3690bf-7851-4700-a156-8bc460c08077.png">

Do let me know what you think or if there are any corner cases missed. Thanks 🙏

Reply:
Oh I realise that you have enforced that the `targetIndexes` must be sorted when initializing the `delete` command and that's why the issue I was talking about would never happen. Thanks for the clarification!